src/context/products.context.jsx

```diff
@@ -0,0 +1,36 @@
+const { createContext, useState, useEffect, useContext } = require("react");
+
+const defaultValue = null;
+const axios = require("axios");
+const ContextOfProducts = createContext(defaultValue);
+
+const ProductProvider = ({ children }) => {
+  const [itemList, setItemList] = useState([]);
+  // initialized with array to store the items
+
+  useEffect(() => {
+    const obtainProductList = async () => {
+      try{
+        const res = await axios.get("/api/products");
+
+        setItemList(res.data.products);
+      }
+   catch(error)
+   {
+     console.error(error)
+   }
+    
+    };
+    obtainProductList();
+  }, []);
```

Comment:
[use this](https://developer.mozilla.org/en-US/docs/Glossary/IIFE) insetead of calling function